js/src/components/paid-ads/budget-section/budget-recommendation/index.js

```diff
@@ -50,17 +49,19 @@ function toRecommendationRange( isMultiple, ...values ) {
 }
 
 const BudgetRecommendation = ( props ) => {
-	const { countryCodes, dailyAverageCost = Infinity } = props;
-	const { data } = useFetchBudgetRecommendationEffect( countryCodes );
+	const { countryCodes, dailyAverageCost = Infinity, data } = props;
 	const map = useCountryKeyNameMap();
 
 	if ( ! data ) {
 		return null;
 	}
 
 	const { currency, recommendations } = data;
-	const { daily_budget: dailyBudget, country } =
-		getHighestBudget( recommendations );
+	const { daily_budget: dailyBudget, country } = getHighestBudget(
+		recommendations.filter( ( recommendation ) =>
```

Comment:
I was able to trigger a case where `recommendations` was initially undefined, which caused this to throw a fatal error:

```
TypeError: Cannot read properties of undefined (reading 'filter')
    at BudgetRecommendation (index.js:61:19)
    at renderWithHooks (react-dom.development.js:15486:18)
    at mountIndeterminateComponent (react-dom.development.js:20098:13)
    at beginWork (react-dom.development.js:21621:16)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at beginWork$1 (react-dom.development.js:27485:7)
    at performUnitOfWork (react-dom.development.js:26591:12)
    at workLoopSync (react-dom.development.js:26500:5)
```

To do so, I changed the selected audience countries in step 2 and then this triggered when I got to step 4.

Reply:
This appears to be fixed.